woocommerce/woocommerce-android
# Issue/11006 show custom date range

Closes: #11006


Don't panic about the number of files changed. I've removed some strings, which generated a lot of small file changes. 

In any case, I'm sorry for the size of this PR. It is big. I intended to do it in several smaller ones, but once I started refactoring the UI layer to replace the usages of `StatsGranularity` with the new model `StatsTimeRangeSelection`, things quickly got out of hand. I wanted to submit a functional compiling version of the feature, so it could be tested. But in the end it resulted in way more changes than I expected.

### Description

Handles the UI for custom range tab:
- Displays the selected custom range on top of the stats
- Allows editing the custom range by tapping on the custom range title
- Handles granularity for custom ranges and displays the granularity value right below the chart
- Tapping on the date range will open the date range picker
- Selecting a different granularity other than custom range should hide the custom range text
- Sliding the finger across the chart when cusom range is selected should display the correct date/time. 

After some discussion with Joe these are some expected changes in the stats view. 

<img width="300"  src="https://github.com/woocommerce/woocommerce-android/assets/2663464/6b235029-b811-4750-9bdd-de66d491209c"> 

**Missing**
- Restoring the selected custom range when killing the app

### Testing instructions

1. Click on on the calendar icon to the right of the tabs in My Store screen
2. Add a custom date range
3. Play around with different custom ranges to verify the granularities for each custom range length are calculated correctly. 
4. Play around with the chart verifying the values displayed make sense
5. Smoke test the rest of the tabs to ensure nothing is broken.

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/2663464/e6d5402b-29e7-4773-a984-29dd50d2941c
